### PR TITLE
Removed unused parameters

### PR DESCRIPTION
**Pull request by Mailaender**
This fixes https://docs.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0060 as I was asked to not randomly drop those in unrelated pull requests I decided to fix them all at once.

### CODE REVIEW

File: OpenRA.Mods.Common/Widgets/Logic/Installation/InstallFromDiscLogic.cs [hunk 1]
```diff
@@ -54,7 +54,7 @@ enum Mode { Progress, Message, List }
 		Mode visible = Mode.Progress;
 
 		[ObjectCreator.UseCtor]
-		public InstallFromDiscLogic(Widget widget, ModContent content, Dictionary<string, ModContent.ModSource> sources, Action afterInstall)
```
Comment (RoosterDragon) [inline OpenRA.Mods.Common/Widgets/Logic/Installation/InstallFromDiscLogic.cs:57]: There's a commented out line with `afterInstall` you can remove. Additionally, it is passed via `WidgetArgs` in `ModContentLogic`. i.e. just search for the string `afterInstall` and nuke it.

File: OpenRA.Platforms.Default/OpenAlSoundEngine.cs [hunk 2]
```diff
@@ -352,16 +352,10 @@ public void SetSoundLooping(bool looping, ISound sound)
...
+			Dispose();
 		}
 
 		public void Dispose()
-		{
-			Dispose(true);
-			GC.SuppressFinalize(this);
-		}
-
-		void Dispose(bool disposing)
```
Comment (RoosterDragon) [inline OpenRA.Platforms.Default/OpenAlSoundEngine.cs:364]: Revert these changes. The correct fix here is that the `disposing` should guard managed cleanup. i.e. we should use it to guard the StopAllSounds call.
```c#
if (disposing)
    StopAllSounds();
```

File: OpenRA.ruleset [hunk 3]
```diff
@@ -13,6 +13,7 @@
 
   <Rules AnalyzerId="Microsoft.CodeAnalysis.CSharp.CodeStyle" RuleNamespace="Microsoft.CodeAnalysis.CSharp.CodeStyle">
     <Rule Id="IDE0005" Action="Warning" /><!-- Using directive is unnecessary. -->
+    <Rule Id="IDE0060" Action="Warning" /><!-- Unused parameter. -->
```
Comment (RoosterDragon) [inline OpenRA.ruleset]: Can we add this to `.editorconfig` instead please?

File: OpenRA.Game/VoiceExts.cs [hunk 4]
```diff
@@ -49,7 +49,7 @@ public static bool HasVoice(this Actor self, string voice)
 			return self.TraitsImplementing<IVoiced>().Any(x => x.HasVoice(self, voice));
 		}
 
-		public static void PlayVoiceForOrders(this World w, Order[] orders)
+		public static void PlayVoiceForOrders(Order[] orders)
```
Comment (RoosterDragon) [inline OpenRA.Game/VoiceExts.cs]: This means this is no longer an extension method and living in `VoiceExts` seems weird.

Should we rename it? Or maybe make it `this Order[] orders` so we can use it as an extension on an array of orders?

File: OpenRA.Mods.Common/Traits/Attack/AttackBase.cs [hunk 5]
```diff
@@ -513,7 +513,7 @@ public bool CanTarget(Actor self, in Target target, List<Actor> othersAtTarget,
 					case TargetType.FrozenActor:
 						return CanTargetActor(self, target, ref modifiers, ref cursor);
 					case TargetType.Terrain:
-						return CanTargetLocation(self, self.World.Map.CellContaining(target.CenterPosition), othersAtTarget, modifiers, ref cursor);
```
Comment (RoosterDragon) [inline OpenRA.Mods.Common/Traits/Attack/AttackBase.cs:516]: Does this mean `othersAtTarget` is now unused?

File: OpenRA.Mods.Common/Traits/Crates/Crate.cs [hunk 6]
```diff
@@ -191,7 +191,7 @@ public void SetPosition(Actor self, WPos pos)
 		// Sets the location (Location) and position (CenterPosition)
 		public void SetPosition(Actor self, CPos cell, SubCell subCell = SubCell.Any)
 		{
-			SetLocation(self, cell, subCell);
```
Comment (RoosterDragon) [inline OpenRA.Mods.Common/Traits/Crates/Crate.cs:194]: Does this mean `subCell` is now unused?

File: OpenRA.Mods.Common/Widgets/ControlGroupsWidget.cs [hunk 7]
```diff
@@ -40,7 +40,7 @@ public class ControlGroupsWidget : Widget
 		public readonly string JumpToGroupKeyPrefix = null;
 
 		[CustomLintableHotkeyNames]
-		public static IEnumerable<string> LinterHotkeyNames(MiniYamlNode widgetNode, Action<string> emitError, Action<string> emitWarning)
+		public static IEnumerable<string> LinterHotkeyNames(MiniYamlNode widgetNode, Action<string> emitError)
```
Comment (RoosterDragon) [inline OpenRA.Mods.Common/Widgets/ControlGroupsWidget.cs:43]: `CustomLintableHotkeyNames` appears to involve reflection so I imagine this would break as-is.

Options:
- Restore the parameter and suppress it
- It seems we've removed `emitWarning` from all sites with this attribute, maybe we should just remove `emitWarning` support from `CheckChromeHotkeys.cs` as not required. We would then keep the parameter removals.

**Review by RoosterDragon — CHANGES_REQUESTED** [review]

File: OpenRA.Mods.Common/Traits/Crates/Crate.cs (hunk 6 above)
Comment (Mailaender) [reply OpenRA.Mods.Common/Traits/Crates/Crate.cs:194]: No, `Mobile` still uses it.

File: OpenRA.Mods.Common/Traits/Attack/AttackBase.cs (hunk 5 above)
Comment (Mailaender) [reply OpenRA.Mods.Common/Traits/Attack/AttackBase.cs:516]: It seemed that it already was unused.

Comment (Mailaender) [reply OpenRA.Mods.Common/Traits/Attack/AttackBase.cs:516]: Oh, you mean in `CanTarget`. Wow, that is all dead code. Removed it.

File: OpenRA.Mods.Common/Widgets/Logic/Installation/InstallFromDiscLogic.cs (hunk 1 above)
Comment (Mailaender) [reply OpenRA.Mods.Common/Widgets/Logic/Installation/InstallFromDiscLogic.cs:57]: Removed it completely.

**Review by RoosterDragon — DISMISSED** [review]

File: OpenRA.Game/Orders/UnitOrderGenerator.cs [hunk 8]
```diff
@@ -46,7 +46,7 @@ public virtual IEnumerable<Order> Order(World world, CPos cell, int2 worldPixel,
 			var target = TargetForInput(world, cell, worldPixel, mi);
 			var actorsAt = world.ActorMap.GetActorsAt(cell).ToList();
```
Comment (abcdefg30) [inline OpenRA.Game/Orders/UnitOrderGenerator.cs]: Seems like this is unused now?

File: OpenRA.Game/Orders/UnitOrderGenerator.cs [hunk 9]
```diff
@@ -78,7 +78,7 @@ public virtual string GetCursor(World world, CPos cell, int2 worldPixel, MouseIn
 			else
 			{
 				var ordersWithCursor = world.Selection.Actors
-					.Select(a => OrderForUnit(a, target, actorsAt, cell, mi))
+					.Select(a => OrderForUnit(a, target, cell, mi))
```
Comment (abcdefg30) [inline OpenRA.Game/Orders/UnitOrderGenerator.cs:79]: Same for `actorsAt` here.

File: OpenRA.Game/Orders/UnitOrderGenerator.cs (hunk 8 above)
Comment (Mailaender) [reply OpenRA.Game/Orders/UnitOrderGenerator.cs]: Yes, nice spot. A small perf win.

**Review by abcdefg30 — APPROVED** [review]

**Comment by penev92** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/74b8febe9298de9d645b8423d15c37ca849bbca4...2a63e85e99105d046e4188ad6656da5ad454c8b3)
